doc/userguide/rules/suri-rule-guide-for-the-snort-expert.rst

```diff
@@ -0,0 +1,906 @@
+references to Suricata apply to version 3.1 and before (where noted).
+
+Contents
+--------
+
+.. contents::
+
+Other References
+----------------
+
+Suricata Documentation:
+
+-  :doc:`index`
+-  :doc:`../index`
+-  :doc:`snort-compatibility`
+
+Automatic Protocol Detection
+----------------------------
+
+-  Suricata does automatic protocol detection of the following
+   application layer protocols:
+
+   -  dcerpc
+   -  dns
+   -  http
+   -  imap (detection only by default; no parsing)
+   -  ftp
+   -  modbus (Suricata v3 and later; disabled by default; minimalist probe parser; can lead to false positives)
+   -  msn (detection only by default; no parsing)
+   -  smb
+   -  smb2 (disabled internally inside the engine)
+   -  smtp
+   -  ssh
+   -  tls (SSLv2, SSLv3 & TLSv1)
+   
+-  In Suricata, protocol detection is port agnostic (in most cases). In
+   Snort, in order for the ``http_inspect`` and other preprocessors to be
+   applied to traffic, it has to be over a configured port.
+
+   -  Some configurations for app-layer in the Suricata yaml can/do by default
+      specify specific destination ports (e.g. DNS)
+   -  **You can look on 'any' port without worrying about the
+      performance impact that you would have to be concerned about with
+      Snort.**
+
+-  If the traffic is detected as HTTP by Suricata, the ``http_*``
+   buffers are populated and can be used, regardless of port(s)
+   specified in the rule.
+
+-  You don't have to check for the http protocol (i.e.
+   ``alert http ...``) to use the ``http_*`` buffers although it
+   is recommended.
+
+-  If you are trying to detect legitimate (supported) application layer
+   protocol traffic and don't want to look on specific port(s), the rule
+   should be written as ``alert <protocol> ...`` with ``any`` in
+   place of the usual protocol port(s).  For example, when you want to
+   detect HTTP traffic and don't want to limit detection to a particular
+   port or list of ports, the rules should be written as
+   ``alert http ...`` with ``any`` in place of
+   ``$HTTP_PORTS``.
+   
+   -  You can also do ``app-layer-protocol:<protocol>;`` inside the rule.
+   
+   So, instead of this Snort rule::
+     
+      alert tcp $HOME_NET any -> $EXTERNAL_NET $HTTP_PORTS ...
+
+   Do this for Suricata::
+    
+      alert http $HOME_NET -> $EXTERNAL_NET any ...
+
+   Or::
+    
+      alert tcp $HOME_NET any -> $EXTERNAL_NET any (app-layer-protocol:http; ...
+
+``content`` Keyword
+-------------------
+
+-  *Older* versions of Suricata (before 3.0) cannot have a ``content`` whose 
+   length greater than 255 characters.
+
+  -  Split up longer content matches if you are running a version of Suricata before 3.0.
+
+-  :doc:`snort-compatibility`
+-  `https://redmine.openinfosecfoundation.org/issues/1281 <https://redmine.openinfosecfoundation.org/issues/1281>`_
+
+``flow`` Keyword
+-------------------
+
+-  Suricata does not support ``not_established`` for the ``flow`` keyword.
```

Comment:
Actually the docs were written for 3.1 (since this is what was/is stable) but I can update for 3.2.